agent/job_runner.go

```diff
@@ -392,6 +394,28 @@ func (r *JobRunner) normalizeVerificationBehavior(behavior string) (string, erro
 	}
 }
 
+func filterEnv(environmentVariables map[string]string, allowedVariables []string) []string {
+	// Filter out if the user has unlisted environment variables
+	// It adds environment variable to denied list and removes if it finds matching regex
+	var deniedVariables []string
```

Comment:
nit: a small performance improvement would be to use make
deniedVariables := make([]string, len(environmentVariables))
Then you avoid resizing it.

Reply:
I tried `make` and it failed when removing the environment variable from the `deniedVariables` list when it got a match, which I think, is because the control logic depends on the list size.